messages/proposed_event.go

```diff
@@ -1,14 +1,108 @@
 package messages
 
 import (
+	"encoding/json"
+
 	uuid "github.com/gofrs/uuid"
 )
 
 // ProposedEvent ...
 type ProposedEvent struct {
-	EventID      uuid.UUID
-	EventType    string
-	ContentType  string
-	Data         []byte
-	UserMetadata []byte
+	eventID      uuid.UUID
+	eventType    string
+	contentType  string
+	data         []byte
+	metadata []byte
+}
+
+func NewJsonProposedEvent(eventType string, payload interface{}) (ProposedEvent, error) {
+	event := ProposedEvent {}
+	bytes, err := json.Marshal(payload)
+
+	if err != nil {
+		return event, err
+	}
+
+	event.eventType = eventType
+	event.contentType = "application/json"
+	event.data = bytes
+	event.metadata = []byte{}
+
+	return event, nil
+}
+
+func NewBinaryProposedEvent(eventType string, bytes []byte) ProposedEvent {
+	event := ProposedEvent {
+		eventType: eventType,
+		contentType: "application/octet-stream",
+		data: bytes,
+		metadata: []byte{},
+	}
+
+	return event
+}
+
+func (event ProposedEvent) GetEventID() uuid.UUID {
```

Comment:
In Go getters should not have GetXXX prefix. More here: https://golang.org/doc/effective_go#Getters

Reply:
(this comment is valid for other getters in that format in this PR)